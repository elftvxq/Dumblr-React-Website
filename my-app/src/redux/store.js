import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import userReducer from './reducers/userReducer';
import dataReducer from './reducers/dataReducer';
import uiReducer from './reducers/uiReducer';

const initialState = {};

const middleware = [thunk];

const reducers = combineReducers({
    user: userReducer,
    data: dataReducer,
    UI: uiReducer
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

// const store = createStore(
//     reducers, 
//     initialState, 
//     compose(applyMiddleware(...middleware), 
//     window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
//   )
// );

const store = createStore(
    reducers,
    initialState,
    compose(applyMiddleware(...middleware),
        composeEnhancers()
    )
);

export default store;