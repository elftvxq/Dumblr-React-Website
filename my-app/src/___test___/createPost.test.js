import React from 'react';
import CreatePostComponent from '../components/post/CreatePost';
import { Provider } from "react-redux";
import renderer, { create} from 'react-test-renderer';
import { createStore } from 'redux';
import { reducers } from '../redux/store';

// enzyme 及 render Component 的 method : shallow
import Enzyme, { shallow, mount  } from 'enzyme'; 
// 將對應 React 版本的解析器導入
import Adapter from 'enzyme-adapter-react-16';

// 以該解析器提供給 Enzyme 做渲染 Component 的設置
Enzyme.configure({ adapter: new Adapter() });


//snapshot test
it('renders correctly', () => {
    let store = createStore(reducers);
    const tree = renderer
        .create( 
        <Provider store={store}>
            <CreatePostComponent />   
        </Provider>
        )
            .toJSON(); expect(tree).toMatchSnapshot();
        });

//使用Enzyme test 
let store = createStore(reducers);
let wrapper;
beforeEach(() => {
     wrapper = mount(
        <Provider store={store}>
            <CreatePostComponent/>   
        </Provider>);
});



describe('<CreatePostComponent> rendering', () => {
     it('should render a <p> to display the number of the post types', () => {
        expect(wrapper.find('p')).toHaveLength(5);
     });
})
