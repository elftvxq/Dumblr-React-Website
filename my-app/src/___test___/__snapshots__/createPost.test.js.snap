// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="post-bar"
>
  <img
    alt=""
    className="avatar-Pic"
  />
  <div
    className="post-section"
  >
    <div
      className="post-text"
      onClick={[Function]}
    >
      <img
        alt="文字"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <p
        className="icon-text"
      >
        文字
      </p>
    </div>
    <div
      className="post-pic"
      onClick={[Function]}
    >
      <img
        alt="連結"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <p
        className="icon-text"
      >
        連結
      </p>
    </div>
    <div
      className="post-quote"
      onClick={[Function]}
    >
      <img
        alt="引述"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <p
        className="icon-text"
      >
        引述
      </p>
    </div>
    <div
      className="post-gif"
      onClick={[Function]}
    >
      <img
        alt="GIF"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <p
        className="icon-text"
      >
        GIF
      </p>
    </div>
    <div
      className="post-video"
      onClick={[Function]}
    >
      <img
        alt="影片"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <p
        className="icon-text"
      >
        影片
      </p>
    </div>
  </div>
</div>
`;
